workflows/analyze.smk

```diff
@@ -155,3 +155,54 @@ rule calculate_rhats_4chains:
         iteration = iteration[3:]
         yg.serialize.save_metric_result(iteration, list(rhat), fp)
 
+
+
+rule calculate_ess_4chains:
+    """Calculate ESS for 4 chains, each with a different init tree and seed but same data and config."""
+    input:
+        mcmc_metric_samples1="{DATADIR}/{experiment}/analysis/MCMC_{mcmc_seed1,\d+}-{mutation_data_id}-i{init_tree_id1}-{mcmc_config_id}/{base_tree_id}/{metric}.json",
+        mcmc_metric_samples2="{DATADIR}/{experiment}/analysis/MCMC_{mcmc_seed2,\d+}-{mutation_data_id}-i{init_tree_id2}-{mcmc_config_id}/{base_tree_id}/{metric}.json",
+        mcmc_metric_samples3="{DATADIR}/{experiment}/analysis/MCMC_{mcmc_seed3,\d+}-{mutation_data_id}-i{init_tree_id3}-{mcmc_config_id}/{base_tree_id}/{metric}.json",
+        mcmc_metric_samples4="{DATADIR}/{experiment}/analysis/MCMC_{mcmc_seed4,\d+}-{mutation_data_id}-i{init_tree_id4}-{mcmc_config_id}/{base_tree_id}/{metric}.json",
+    wildcard_constraints:
+        mutation_data_id = "CS.*",
+        mcmc_config_id= "MC_(?:(?!/).)+",
```

Comment:
I'd find a comment explaining what this regex is supposed to catch very useful :slightly_smiling_face: